### DISCUSSION TITLE
Update script nocoDB to version 0.202.5

### DISCUSSION THREAD

**Discussion by epcos71** [tteck/Proxmox#1949] [Help Needed]
Hi,
I just started the Helper Script for nocoDB to update from 0.202.4 to 0.202.5. The script worked without error.
After calling the url my browser told me that the site is not reachable.

Then I checked the LXC and saw that there is heavy load on the cpu. After waitung for 60 minutes, the load was still up to 100% and I had still no access to nocodb.

After restoring the old version, I made the manual commands of the script:

1. npm uninstall -s --save nocodb
2. npm install -s --save nocodb
3. systemctl restart nocodb.service

All went fine, so I think the problem is after the service hast started.
Has anybody the same experience and could fix it?

Thanks in advance

**Comment by tteck** [comment]
It seems that the installation and update process for the NodeJS Application is currently not functioning as expected. I've made some improvements to the script, which now installs the binary. This change will simplify the installation and updating process.

**Reply by epcos71** [reply]
Thanks for the fast reply!

I tried to update now with the updated script, but now the script interrupts with the rm command...
 
> Updating NocoDB...rm: cannot remove 'nocodb': No such file or directory
>
>[ERROR] in line 59: exit code 1: while executing command rm nocodb
>

**Reply by tteck** [reply]
Correct, everything has changed, meaning a fresh NocoDB LXC is required.

**Reply by epcos71** [reply]
ok, then there is no possibility to migrate my data, right?

**Reply by tteck** [reply]
I really don't know. Are there any `.db` files in `/opt/nocodb`? Maybe back those up and restore them in the new `/opt/nocodb` folder.

**Reply by tteck** [reply]
hold on just a minute, I'll try and see if I can make the new update switch to the binary for the old install if you're willing to test.

**Reply by epcos71** [reply]
sure ;)

**Reply by tteck** [reply]
Ok, try to update now. 🤞🏻

**Reply by epcos71** [reply]
Thanks! The update procedure started and ended with no errors ("Updated Succesfully"), BUT the version did not change to 0.202.5 (within account setting there is still the button "Upgrade available" visible)...

**Reply by tteck** [reply]
reboot the LXC

**Reply by epcos71** [reply]
done... also cleared the browsers cache :)

**Reply by tteck** [reply]
if you `cat /etc/systemd/system/nocodb.service` does this line `ExecStart=/opt/nocodb/./nocodb` exist?

**Reply by tteck** [reply]
found an error, fixed. Try updating again.

**Reply by tteck** [reply]
The `/etc/systemd/system/nocodb.service` file should be

```
[Unit]
Description=nocodb

[Service]
Type=simple
Restart=always
User=root
WorkingDirectory=/opt/nocodb
ExecStart=/opt/nocodb/./nocodb

[Install]
WantedBy=multi-user.target
```

**Reply by tteck** [reply]
That will use the new binary

**Reply by epcos71** [reply]
First of all thanks a lot for the support and your time!!!!

First I changed the nocodb.service file as mentioned, then I started the script again. It interrupted with the following error:
```
Loading...
bash: line 88: warning: here-document at line 87 delimited by end-of-file (wanted `EOF')
bash: -c: line 89: syntax error: unexpected end of file
```

**Reply by tteck** [reply]
Sounds like a Proxmox cache issue, did you try updating via the script?

**Reply by tteck** [reply]
testing...

**Comment by epcos71** [comment]
> Sounds like a Proxmox cache issue, did you try updating via the script?

Yes, I used the script within the LXC shell...

**Reply by tteck** [reply]
Ok, try updating again. You may see
```
 - Updating NocoDB...Warning: The unit file, source configuration file or drop-ins of nocodb.service changed on disk. Run 'systemctl daemon-reload' to reload units.
 ```
but it will execute `systemctl daemon-reload`

**Reply by epcos71** [reply]
Script ended with "Updated successfully", but version is still not 0.202.5 (Upgrade available). Container was rebooted and the browser's cache was cleared...

**Reply by tteck** [reply]
Well, we tried. Create a new NocoDB Lxc.

**Reply by epcos71** [reply]
Yes, we did! Thanks a lot for your time and support! ;)

**Reply by epcos71** [reply]
But I just saw that there was only 1 update of the script 2 hours ago... ?

**Reply by tteck** [reply]
Not following

**Reply by tteck** [reply]
![Screenshot 2023-10-21 12 29 24 PM](https://github.com/tteck/Proxmox/assets/21342670/223382c9-d97b-4624-9dde-5f789c8a879f)
